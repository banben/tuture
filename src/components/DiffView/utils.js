function computeOldLineNumber(change) {
  if (change.isInsert) {
    return -1;
  }

  return change.isNormal ? change.oldLineNumber : change.lineNumber;
}

function computeNewLineNumber(change) {
  if (change.isDelete) {
    return -1;
  }

  return change.isNormal ? change.newLineNumber : change.lineNumber;
}

function getChangeKey(change) {
  if (!change) {
    throw new Error('change is not provided');
  }

  const {
    isNormal,
    isInsert,
    lineNumber,
    oldLineNumber,
  } = change;

  if (isNormal) {
    return 'N' + oldLineNumber;
  }

  const prefix = isInsert ? 'I' : 'D';
  return prefix + lineNumber;
}

export {
  computeOldLineNumber,
  computeNewLineNumber,
  getChangeKey,
}
