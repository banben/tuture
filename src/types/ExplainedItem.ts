type ExplainType = 'pre' | 'post';
type EditMode = 'edit' | 'preview';

type ToolType =
  | 'b'
  | 'i'
  | 'h'
  | 'list'
  | 'numbered list'
  | 'blockquotes'
  | 'link'
  | 'img'
  | 'code'
  | 'block code';

export { ExplainType, EditMode, ToolType };
