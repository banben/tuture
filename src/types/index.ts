interface ChangedFile {
  file: string;
  explain: string;
  collapse?: boolean;
}

interface TutureMeta {
  name: string;
  language: string;
  version: string;
  description: string;
  maintainer: string;
  topics: string[];
}

interface Commit {
  name: string;
  commit: string;
}

interface Step extends Commit {
  explain: string;
  diff: ChangedFile[];
}

interface Tuture extends TutureMeta {
  steps: Step[];
}

interface Change {
  content: string;
  type: string;
  isInsert?: boolean;
  isNormal?: boolean;
  isDelete?: boolean;
  lineNumber?: number;
  oldLineNumber?: number;
  newLineNumber?: number;
}

interface Hunk {
  content: string;
  newLines: number;
  oldLines: number;
  oldStart: number;
  newStart: number;
  changes: Change[];
  isPlain?: boolean;
}

export { ChangedFile, TutureMeta, Commit, Step, Tuture, Change, Hunk };
