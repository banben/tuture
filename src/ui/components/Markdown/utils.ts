const spliceStr = (str = '', typeStr: string, start: number, end: number) => {
  return str
    .slice(0, start)
    .concat(
      typeStr,
      str.slice(start, end),
      typeStr,
      str.slice(end, str.length),
    );
};

const insertStr = (str = '', typeStr: string, insertPosition: number) => {
  return str
    .slice(0, insertPosition)
    .concat(typeStr, str.slice(insertPosition, str.length));
};

export { spliceStr, insertStr };
