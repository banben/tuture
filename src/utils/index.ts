import * as obj from './object';
import * as arr from './array';

const tutureUtilities = {
  ...arr,
  ...obj,
};

export default tutureUtilities;
