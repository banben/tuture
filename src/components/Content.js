import React from 'react';

import { isArray } from './utils/';
import ContentItem from './ContentItem';
import './css/Content.css';



export default class Content extends React.Component {
  renderExplain = (explain) => {
    return (
      isArray(explain)
      ? (explain.map((explainItem, i) => <p key={i}>{explainItem}</p>))
      : <p>{explain}</p>
    );
  }

  render() {
    const { name, explain, diff, commit } = this.props.content;
    return (
      <div className="Content">
        <div className="Content-header">
          <h1>{name}</h1>
          <button
            onClick={this.props.changeViewType}
          >
            {this.props.viewType}
          </button>
        </div>
        {this.renderExplain(explain)}
        <ContentItem
          diff={diff}
          commit={commit}
          viewType={this.props.viewType}
          renderExplain={this.renderExplain}
        />
      </div>
    );
  }
}
