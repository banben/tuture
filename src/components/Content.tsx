import React from 'react';
import styled from 'styled-components';
import { inject, observer } from 'mobx-react';

import { AppProps } from './App';
import StepContent from './StepContent';
import { Tuture } from '../types/';
import Store from './store';

interface ContentProps extends AppProps {}

/* tslint:disable-next-line */
const ContentWrapper = styled.div`
  margin-top: -32px;
`;

@inject('store')
@observer
export default class Content extends React.Component<ContentProps> {
  public render() {
    const { store, diff } = this.props;
    const renderContent: any = [];

    store.tuture.steps.map((step, index: number) => {
      const diffItem = diff[index];

      if (!step.outdated) {
        renderContent.push(
          <StepContent
            key={index}
            content={step}
            index={index}
            diffItem={diffItem}
          />,
        );
      }
    });
    return <ContentWrapper>{renderContent}</ContentWrapper>;
  }
}
