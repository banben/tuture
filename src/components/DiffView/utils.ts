import { Change as ChangeType } from '../../types';

function computeOldLineNumber(change: ChangeType): number {
  if (change.isInsert) {
    return -1;
  }

  return change.isNormal ? change.oldLineNumber : change.lineNumber;
}

function computeNewLineNumber(change: ChangeType): number {
  if (change.isDelete) {
    return -1;
  }

  return change.isNormal ? change.newLineNumber : change.lineNumber;
}

function getChangeKey(change: ChangeType): string {
  if (!change) {
    throw new Error('change is not provided');
  }

  const {
    isNormal,
    isInsert,
    lineNumber,
    oldLineNumber,
  } = change;

  if (isNormal) {
    return 'N' + oldLineNumber;
  }

  const prefix = isInsert ? 'I' : 'D';
  return prefix + lineNumber;
}

export {
  computeOldLineNumber,
  computeNewLineNumber,
  getChangeKey,
}
