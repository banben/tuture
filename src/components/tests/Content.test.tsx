import React from 'react';
import renderer from 'react-test-renderer';

import Content from '../Content';
import tutureUtilities from '../../utils/index';
import { Step } from '../../types/index';
import { diff } from './utils/hunkData';

test('Content render successfully', () => {
  const content = {
    name: 'my first commit',
    commit: '1bd58a3',
    explain: 'tuture is cool ? yes! absolutly',
    diff: diff,
  };

  let viewType = 'unified';
  const changeViewType = (): void => {
    viewType = viewType === 'unified' ? 'split' : 'unified';
  };

  const component = renderer.create(
    <Content
      key={'content'}
      content={content}
      viewType={'unified'}
      changeViewType={changeViewType}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
