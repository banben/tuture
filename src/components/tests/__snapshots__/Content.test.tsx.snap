// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content render successfully 1`] = `
<div
  className="sc-bwzfXH jUaPTe"
>
  <div
    className="sc-htpNat huQAjV"
  >
    <h1>
      my first commit
    </h1>
    <button
      onClick={[Function]}
    >
      unified
    </button>
  </div>
  <p>
    tuture is cool ? yes! absolutly
  </p>
  <div
    className="ContentItem"
  />
</div>
`;
